libcnb-test/src/build.rs

```diff
@@ -1,62 +1,138 @@
         cargo_profile,
-        &cargo_env,
         target_triple.as_ref(),
-    )
-    .map_err(PackageCrateBuildpackError::BuildBinariesError)?;
+    );
+
+    // TODO: this could accidentally detect a packaged meta-buildpack twice. how should we ignore directories
+    //       containing packaged buildpacks that might get detected now that the user controls --package-dir?
+    //       - support .gitignore or some other persistent configuration?
+    //       - during packaging should we create some type of file that indicates the output is a packaged buildpack?
+    let buildpack_dependency_graph =
+        build_libcnb_buildpacks_dependency_graph(&workspace_root_path, &[])
+            .map_err(PackageTestBuildpackError::CreateBuildpackDependencyGraph)?;
 
-    assemble_buildpack_directory(
-        buildpack_dir.path(),
-        cargo_manifest_dir.join("buildpack.toml"),
-        &buildpack_binaries,
+    let root_node = buildpack_dependency_graph
+        .node_weights()
+        .find(|node| node.buildpack_id == buildpack_id.clone());
+
+    assert!(
+        root_node.is_some(),
+        "Could not package directory as buildpack! No buildpack with id `{buildpack_id}` exists in the workspace at {}",
+        workspace_root_path.display()
+    );
+
+    let build_order = get_dependencies(
+        &buildpack_dependency_graph,
+        &[root_node.expect("The root node should exist")],
     )
-    .map_err(PackageCrateBuildpackError::CannotAssembleBuildpackDirectory)?;
+    .map_err(PackageTestBuildpackError::GetDependencies)?;
+
+    let mut packaged_buildpack_dirs = BTreeMap::new();
+    for node in &build_order {
+        let buildpack_destination_dir = buildpack_dir_resolver(&node.buildpack_id);
+
+        fs::create_dir_all(&buildpack_destination_dir).unwrap();
+
+        libcnb_package::package::package_buildpack(
+            &node.path,
+            cargo_profile,
+            target_triple.as_ref(),
+            &cargo_build_env,
+            &buildpack_destination_dir,
+            &packaged_buildpack_dirs,
+        )
+        .map_err(PackageTestBuildpackError::PackageBuildpack)?;
+
+        packaged_buildpack_dirs.insert(node.buildpack_id.clone(), buildpack_destination_dir);
+    }
+
+    Ok(PackagedBuildpackDir {
+        _root: package_dir,
+        path: buildpack_dir_resolver(buildpack_id),
+    })
+}
 
-    Ok(buildpack_dir)
+pub(crate) struct PackagedBuildpackDir {
+    _root: TempDir,
+    pub(crate) path: PathBuf,
 }
 
 #[derive(Debug)]
-pub(crate) enum PackageCrateBuildpackError {
-    BuildBinariesError(BuildBinariesError),
-    CannotAssembleBuildpackDirectory(std::io::Error),
+pub(crate) enum PackageTestBuildpackError {
     CannotCreateBuildpackTempDirectory(std::io::Error),
     CannotDetermineCrateDirectory(std::env::VarError),
-    CargoMetadataError(cargo_metadata::Error),
+    CannotReadBuildpack(TomlFileError),
+    CreateBuildpackDependencyGraph(BuildBuildpackDependencyGraphError),
```

Comment:
Same as below, if there is a good reason to name this "create" instead of "build" we should rename `BuildBuildpackDependencyGraphError`. If there is no reason, the enum variant should follow the existing naming.

Reply:
renamed to `BuildBuildpackDependencyGraph`